### ISSUE TITLE
HyperMerchant Shops not opening GUI

### ISSUE THREAD

**Issue by brooklynstarr** [Pingger/HyperConomy#1]
Hi,

Since the update to 1.13, I've been unable to open up any of my existing shop GUI's. Initially, only certain shops were failing to open, however now all of them do not open. Do I need to recreate the shops, or is this an issue between hyperconomy and hypermerchant?

Thanks.

**Comment by brooklynstarr** [comment]
Hi, apologies I've been offline for a while. I'm using all the bleeding edge versions of Citizens, Hyperconomy, hypermerchant, Spigot, and Vault. I update Spigot daily. The citizens/hypermerchant versions I get from here: http://ci.citizensnpcs.co/ and Spigot I download/build every day or so. Vault, I use the latest version (1.7.1) found here: https://ci.md-5.net/job/Vault/ and I use your version of hyperconomy, downloaded from your server. I'll update this section in a few when I get a chance to log in to the server, I don't believe there's a stack trace, but I'll double check.

Thanks.

**Comment by brooklynstarr** [comment]
OK- you can close this. Looking at the stack trace, the problem is with HyperMerchant, not Hyperconomy. What plugins do you recommend we use for Hyperconomy shops? Preferably one that is Citizens compatible, though it doesn't have to be. I just want my players to access the shops, and if I can do something temporary until this is fixed it would be appreciated.

Thanks.

Could not pass event NPCRightClickEvent to Citizens v2.0.24-SNAPSHOT (build 1574)
org.bukkit.event.EventException: null
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:308) ~[spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:500) ~[spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:485) ~[spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at net.citizensnpcs.EventListen.onPlayerInteractEntity(EventListen.java:431) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_181]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_181]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_181]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_181]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:304) ~[spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:62) ~[spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:500) ~[spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:485) ~[spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at net.minecraft.server.v1_13_R2.PlayerConnection.a(PlayerConnection.java:1828) ~[spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at net.minecraft.server.v1_13_R2.PacketPlayInUseEntity.a(SourceFile:69) ~[spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at net.minecraft.server.v1_13_R2.PacketPlayInUseEntity.a(SourceFile:13) ~[spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at net.minecraft.server.v1_13_R2.PlayerConnectionUtils.a(SourceFile:10) ~[spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_181]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_181]
	at net.minecraft.server.v1_13_R2.SystemUtils.a(SourceFile:199) [spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at net.minecraft.server.v1_13_R2.MinecraftServer.b(MinecraftServer.java:900) [spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at net.minecraft.server.v1_13_R2.DedicatedServer.b(DedicatedServer.java:417) [spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at net.minecraft.server.v1_13_R2.MinecraftServer.a(MinecraftServer.java:835) [spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at net.minecraft.server.v1_13_R2.MinecraftServer.run(MinecraftServer.java:733) [spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_181]
Caused by: java.lang.NullPointerException
	at grokswell.hypermerchant.ShopMenu.UpdateSortingIcon(ShopMenu.java:314) ~[?:?]
	at grokswell.hypermerchant.ShopMenu.<init>(ShopMenu.java:96) ~[?:?]
	at grokswell.hypermerchant.HyperMerchantTrait.onRightClick(HyperMerchantTrait.java:366) ~[?:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_181]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_181]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_181]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_181]
	at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:304) ~[spigot-1.13.1.jar:git-Spigot-f6a273b-68588da]
	... 24 more
[14:59:22]

**Comment by Pingger** [comment]
It seems, that HyperConomy is unrelated to this issue, since there is no call to anything within the `regalowl.hyperconomy` package, only `grokswell.hypermerchant`. 
My guess would be, that the new Spigot API is not correctly implemented to the HyperMerchant plugin, which causes invalid accesses to items, that got renames (internally. e.g. WOOD is split into ACACIA_WOOD, BIRCH_WOOD, ..., SPRUCE_WOOD; making the WOOD call invalid and possibly causing an exception)

**Comment by Pingger** [comment]
It seems it is looking for the material for the `help5`-Button


```
[...]
Caused by: java.lang.NullPointerException
    at grokswell.hypermerchant.ShopMenu.UpdateSortingIcon(ShopMenu.java:314) ~[?:?]
[...]
```
Looking at that line in the Source:
https://github.com/remremrem/HyperMerchant/blob/46be75c612a834496b836e0bb53fa5f860e6af8d/src/main/java/grokswell/hypermerchant/ShopMenu.java#L314
And following it; Makes me find this piece of code, so you should check the config for some invalid item data.
```
    	help5 = new ItemStack(Material.getMaterial(menuButtonData.getString("help5.material")));
    	help5.setDurability(Short.parseShort(menuButtonData.getString("help5.data")));
    	setItemNameAndLore(help5, menuButtonData.getString("help5.name"), ParseLore(menuButtonData.getString("help5.lore")));
```

**Comment by brooklynstarr** [comment]
Ok, I did find a menu button config YML that referred to that particular item, help5. It was set to "BOOK" and I saw this hadn't changed from what it was originally in Minecraft. However the other items, Stained Glass, had, so I updated them. Unfortunately, it still points to this help5 sort icon function even if I change the material to something else. I took a close look at the code but unfortunately wasn't able to come up with anything no matter what I tried. 

There's definitely an issue with menu display from a functional perspective and I unfortunately don't know enough about the code to fix it. I've opened up this issue on the Hypermerchant GitHub page but it looks like the project hasn't been updated in a long time. I guess time will tell if anything. I've got the code to play with here and if I have more time i'll take a deeper look.

Thanks for your help, much appreciated.
